common/djangoapps/cors_csrf/middleware.py

```diff
@@ -22,35 +47,76 @@
 
 from django.conf import settings
 from django.middleware.csrf import CsrfViewMiddleware
+from django.core.exceptions import MiddlewareNotUsed, ImproperlyConfigured
 
 log = logging.getLogger(__name__)
 
 
+def is_cross_domain_request_allowed(request):
+    """Check whether we should allow the cross-domain request.
+
+    We allow a cross-domain request only if:
+
+    1) The request has an http referer header.
+    2) The request is made securely and the referer has "https://" as the protocol.
+    3) The referer domain has been whitelisted.
+
+    Arguments:
+        request (HttpRequest)
+
+    Returns:
+        bool
+
+    """
+    referer = request.META.get('HTTP_REFERER')
+    if not referer:
+        log.debug("Request does not have a referer, so it is not a cross-domain request.")
+        return False
+    referer_parts = urlparse.urlparse(referer)
+
+    domain_is_whitelisted = (
+        getattr(settings, 'CORS_ORIGIN_ALLOW_ALL', False) or
+        referer_parts.hostname in getattr(settings, 'CORS_ORIGIN_WHITELIST', [])
+    )
+    if not domain_is_whitelisted:
+        log.debug(
```

Comment:
I left these as debug because they're likely to create a *lot* of log entries.  (Any request from courses.edx.org, for instance, won't be on the whitelist).  Is this okay?